### PR TITLE
Dynamically changing tooltipBgColor

### PR DESCRIPTION
**Pull request by xvrh**
This PR is to replace this one: https://github.com/imaNNeo/fl_chart/pull/1348.

All credits goes to @apekshamehta

### Original description:

**Changes Made:**

In the bar_chart_data.dart file:

Added a new parameter getTooltipColor to the constructor of BarTouchTooltipData.
Defined a new type GetBarTooltipColor, which is a function that takes a BarChartGroupData parameter and returns a Color.
In the bar_chart_painter.dart file:

Assigned the tooltip color to the _bgTouchTooltipPaint.color property using the getTooltipColor function from the tooltipData object.
Additionally, the getTooltipColor function has been utilized in the bar_chart_sample1.dart file within the feature/dynamic-tooltip-bgcolor branch.

Please review the changes for this pull request.

### CODE REVIEW

File: example/lib/presentation/samples/bar/bar_chart_sample1.dart [hunk 1]
```diff
@@ -162,7 +162,19 @@ class BarChartSample1State extends State<BarChartSample1> {
     return BarChartData(
       barTouchData: BarTouchData(
         touchTooltipData: BarTouchTooltipData(
-          tooltipBgColor: Colors.blueGrey,
+          getTooltipColor: (group) {
+            Color bgColor = switch (group.x) {
+              0 => AppColors.contentColorPurple.withAlpha(100),
```
Comment (imaNNeo) [inline example/lib/presentation/samples/bar/bar_chart_sample1.dart:167]: These colors are not consistent with the design. Please use the Colors.blueGrey just like before

File: example/lib/presentation/samples/line/line_chart_sample3.dart [hunk 2]
```diff
@@ -206,7 +206,22 @@ class _LineChartSample3State extends State<LineChartSample3> {
                     }).toList();
                   },
                   touchTooltipData: LineTouchTooltipData(
-                    tooltipBgColor: widget.tooltipBgColor,
+                    getTooltipColor: (touchedSpot) {
+                      Color bgColor = switch ((
+                        touchedSpot.x.toInt(),
+                        touchedSpot.y.toDouble()
+                      )) {
+                        (0, 1.3) => AppColors.contentColorPurple.withAlpha(100),
```
Comment (imaNNeo) [inline example/lib/presentation/samples/line/line_chart_sample3.dart:214]: Also here, please use the previous color

File: pubspec.yaml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 name: fl_chart
 description: A highly customizable Flutter chart library that supports Line Chart, Bar Chart, Pie Chart, Scatter Chart, and Radar Chart.
-version: 0.66.2
+version: 0.67.0
```
Comment (imaNNeo) [inline pubspec.yaml:3]: You should not upgrade the version. Please read the [contributing](https://github.com/imaNNeo/fl_chart/blob/main/CONTRIBUTING.md) guidelines carefully

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,7 @@
+## 0.67.0
```
Comment (imaNNeo) [inline CHANGELOG.md:1]: You don't need to mention a number here. Just leave it like `## newVerion`, then I will bump it up.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,7 @@
...
+* **FEATURE**  (by @apekshamehta) Added new method called getTooltipColor for axis charts (bar,line,scatter) to change backgroud color of tooltip dynamically.[issue](https://github.com/imaNNeo/fl_chart/issues/1279).
```
Comment (imaNNeo) [inline CHANGELOG.md:2]: Typo in `backgroud` word

**Comment by imaNNeo** [comment]
There are some minor changes. Please fix them, then It is ready to be merged!

File: example/lib/presentation/samples/bar/bar_chart_sample1.dart (hunk 1 above)
Comment (xvrh) [reply example/lib/presentation/samples/bar/bar_chart_sample1.dart]: Done

File: example/lib/presentation/samples/line/line_chart_sample3.dart (hunk 2 above)
Comment (xvrh) [reply example/lib/presentation/samples/line/line_chart_sample3.dart]: Done

File: example/lib/presentation/samples/scatter/scatter_chart_sample2.dart [hunk 6]
```diff
@@ -126,7 +126,23 @@ class _ScatterChartSample2State extends State {
                       : SystemMouseCursors.click;
                 },
                 touchTooltipData: ScatterTouchTooltipData(
-                  tooltipBgColor: Colors.black,
+                  getTooltipColor: (ScatterSpot touchedBarSpot) {
+                    Color bgColor = switch ((
+                      touchedBarSpot.x.toInt(),
+                      touchedBarSpot.y.toInt()
+                    )) {
+                      (4, 4) => AppColors.contentColorPurple.withAlpha(100),
```
[the remark this answers is not in the document]
Comment (xvrh) [reply example/lib/presentation/samples/scatter/scatter_chart_sample2.dart]: Done

File: pubspec.yaml (hunk 3 above)
Comment (xvrh) [reply pubspec.yaml]: Done

File: CHANGELOG.md (hunk 4 above)
Comment (xvrh) [reply CHANGELOG.md]: Done

File: CHANGELOG.md (hunk 5 above)
Comment (xvrh) [reply CHANGELOG.md]: Done

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,7 @@
...
+* **BREAKING** (by @apekshamehta) Removed tooltipBgColor property from Bar, Line and Scatter Charts.
```
[the remark this answers is not in the document]
Comment (xvrh) [reply CHANGELOG.md:4]: Added `repo_files/documentations/migration_guides/0.67.0/MIGRATION_00_67_00.md`. I let you re-organise as you wish.

**Comment by imaNNeo** [comment]
Thanks for your quick response and action!
I hope to see more from you xD

**Pull request merged**